src/AnimateText.js

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import SplitText from '@moxy/react-split-text';
+import IntersectionObserver from '@researchgate/react-intersection-observer';
+import 'intersection-observer'; // Polyfill to support older browsers
+
+const AnimateText = ({
+    children,
+    className,
+    separator,
+    withObserver,
+    shouldShow,
+    transitionDuration,
+    initialDelay,
+    lineDelay,
+    wordDelay,
+}) => {
+    const containerRef = useRef(null);
+    const [isReady, setIsReady] = useState(false);
+    const [shouldAnimate, setShouldAnimate] = useState(withObserver ? false : shouldShow);
+
+    useEffect(() => {
+        const elements = [...containerRef.current.children];
+        let prevOffset = 0;
+        let delay = initialDelay;
+
+        elements.forEach((word, index) => {
```

Comment:
Can't we use `React.Children.forEach`?

Reply:
`React.Children` doesn't work here because we want to target the `containerRef.current.children` that already has the text split.

Reply:
👍